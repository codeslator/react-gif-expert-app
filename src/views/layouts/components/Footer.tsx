import { FC } from 'react';
import { Button } from 'primereact/button';

const Footer: FC = () => {
  return (
    <footer className="surface-0">
      <div className="w-full mx-auto p-4">
        <div className="flex flex-column sm:flex-row align-items-center justify-content-between">
          <a href="https://codeslator.dev/" className="flex align-items-center mb-4 sm:mb-0 no-underline">
            <img src="https://codeslator.dev/wp-content/uploads/2024/07/Iso_Color_White.svg" className="w-3rem h-3rem" alt="Codeslator Logo" />
            <span className="self-center text-2xl font-semibold whitespace-nowrap mx-3 text-white">GifExpert</span>
          </a>
          <div className="w-full sm:w-auto flex flex-row justify-content-evenly align-items-center">
            <Button
              icon="pi pi-facebook"
              severity="secondary"
              text
              aria-label="Facebook"
              size="small"
              className="sm:mr-3"
              link
              onClick={() => window.open('https://www.facebook.com/codeslator', '_blank')}
            />
            <Button
              icon="pi pi-instagram"
              severity="secondary"
              text
              aria-label="Instagram"
              size="small"
              className="sm:mr-3"
              link
              onClick={() => window.open('https://www.instagram.com/codeslator/', '_blank')}
            />
            <Button
              icon="pi pi-linkedin"
              severity="secondary"
              text
              aria-label="LinkedIn"
              size="small"
              className="sm:mr-3"
              link
              onClick={() => window.open('https://www.linkedin.com/in/andres-melendez-dev/', '_blank')}
            />
            <Button
              icon="pi pi-github"
              severity="secondary"
              text
              aria-label="Github"
              size="small"
              className="sm:mr-3"
              link
              onClick={() => window.open('https://www.linkedin.com/in/andres-melendez-dev/', '_blank')}
            />
          </div>
        </div>
        <hr className="my-3 sm:mx-auto border-gray-700 lg:my-4" />
        <span className="block text-sm text-white text-center">
          Â© {new Date().getFullYear()} <a href="https://codeslator.dev/" className="no-underline hover:underline text-primary-500">CodeslatorDev</a>. All Rights Reserved.
        </span>
      </div>
    </footer>
  );
};

export default Footer;