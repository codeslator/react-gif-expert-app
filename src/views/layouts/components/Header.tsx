import { FC } from 'react';
import { Menubar } from 'primereact/menubar';
import { useNavigate } from 'react-router-dom';


const Header: FC = () => {
  const navigate = useNavigate();
  const items = [
    {
      label: 'Home',
      icon: 'pi pi-home',
      command: () => {
        navigate('/');
      }
    },
    {
      label: 'Search GIFs',
      icon: 'pi pi-images',
      command: () => {
        navigate('/gifs');
      }
    },

  ];

  return (
    <Menubar model={items} className="border-none border-noround surface-0" />
  );
};

export default Header;