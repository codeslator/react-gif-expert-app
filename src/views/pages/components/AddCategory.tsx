import { ChangeEvent, FC, FormEvent, useState } from 'react';
import { Button } from 'primereact/button';
import { InputText } from 'primereact/inputtext';

interface AddCategoryProps {
  categories: string[];
  onNewCategory: (value: string) => void;
  onResetCategories: () => void;
}

const AddCategory: FC<AddCategoryProps> = ({ onNewCategory, onResetCategories, categories }) => {
  const [search, setSearch] = useState<string>('');

  const handleSearch = (event: ChangeEvent<HTMLInputElement>) => {
    const { value } = event.target;
    setSearch(value);
  };

  const handleSubmit = (event: FormEvent) => {
    event.preventDefault();
    if (search.trim().length <= 1) return;
    onNewCategory(search.trim());
    setSearch('');
  };

  return (
    <form onSubmit={handleSubmit} aria-label="form">
      <div className="p-inputgroup flex-1">
        {(categories.length > 0) && (
          <Button
            label="Reset"
            icon="pi pi-times"
            severity="danger"
            onClick={onResetCategories}
          />
        )}
        <InputText
          type="text"
          placeholder="Search a category"
          value={search}
          onChange={handleSearch}
        />
        <Button
          label="Search"
          icon="pi pi-search"
          type="submit"
        />
      </div>
    </form>
  );
};

export default AddCategory;