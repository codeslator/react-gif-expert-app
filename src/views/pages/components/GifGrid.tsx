import { FC } from 'react';
import { GifItem, GridSkelleton } from './';
import { useGifs } from '../../../hooks';
import { Gif } from '../../../global';

interface GifGridProps {
  category: string;
}

const GifGrid: FC<GifGridProps> = ({ category }) => {
  const { gifs, isLoading } = useGifs(category);

  return (
    <div>
      <h1>{category}</h1>
      {isLoading ? (
        <GridSkelleton />
      ) : (
        <div className="grid">
          {gifs.map((gif: Gif) => (
            <div key={gif.id} className="col-12 md:col-4 lg:col-2">
              <GifItem gif={gif} />
            </div>
          ))}
        </div>
      )}
    </div>
  );
};

export default GifGrid;