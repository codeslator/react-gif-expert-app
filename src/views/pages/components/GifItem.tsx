import { FC } from 'react';
import { Card } from 'primereact/card';
import { Gif } from '../../../global';

interface GifItemProps {
  gif: Gif;
}

interface ItemHeaderProps {
  title: string;
  url: string;
}

const ItemHeader = ({ title, url }: ItemHeaderProps) => {
  return (
    <img alt={title} src={url} />
  )
};

const GifItem: FC<GifItemProps> = ({ gif }) => {
  return (
    <Card
      title={gif.title}
      header={(
        <ItemHeader title={gif.title} url={gif.url} />
      )}
      className="h-full"
    />
  );
};

export default GifItem;