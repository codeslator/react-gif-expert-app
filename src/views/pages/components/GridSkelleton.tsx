import { FC } from 'react';
import { ItemSkelleton } from './';

const GridSkelleton: FC = () => {
  return (
    <div className="grid">
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
      <ItemSkelleton />
    </div>
  );
};

export default GridSkelleton;