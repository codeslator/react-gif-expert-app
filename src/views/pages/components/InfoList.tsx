import { FC } from 'react';

interface InfoListProps {
  items: string[];
}

const InfoList: FC<InfoListProps> = ({ items }) => {
  return (
    <ul className="list-none p-0">
      {items.map((item: string, index: number) => (
        <li key={index} className="flex align-items-center mb-2">
          <i className="pi pi-check-circle mr-2 text-primary" style={{ fontSize: '1.25rem' }}></i>
          {item}
        </li>
      ))}
    </ul>
  );
};

export default InfoList;