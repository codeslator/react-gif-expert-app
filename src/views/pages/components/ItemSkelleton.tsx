import { FC } from 'react';
import { Skeleton } from 'primereact/skeleton';

const ItemSkelleton: FC = () => {
  return (
    <div className="col-12 md:col-4 lg:col-2 p-2">
      <div className="w-full">
        <Skeleton height="2rem" className="mb-2" />
        <Skeleton height="10rem" className="mb-2" />
        <Skeleton height="6rem" className="mb-2" />
      </div>
    </div>
  );
};

export default ItemSkelleton;