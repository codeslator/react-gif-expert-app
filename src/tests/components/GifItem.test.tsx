import { render, screen } from "@testing-library/react"
import { GifItem } from "../../views/pages/components"

describe('Test <GifItem /> component.', () => {
  const gif = {
    title: 'Pokemon',
    url: 'https://pokemon.com/bulbasaur.jpg',
    id: "abc123"
  }
  
  test('Should to match with the snapshot.', () => {
    const { container } = render(<GifItem gif={gif} />);
    expect(container).toMatchSnapshot();
  });

  test('Should display alt and url correctly.', () => {
    render(<GifItem gif={gif} />);
    // expect(screen.getByRole<HTMLImageElement>('img').src).toBe(gif.url);
    const { src, alt } = screen.getByRole<HTMLImageElement>('img');
    expect(src).toBe(gif.url);
    expect(alt).toBe(gif.title);
  });

  test('Should display the title in the component.', () => {
    render(<GifItem gif={gif} />);
    expect(screen.getByText(gif.title)).toBeTruthy();
  })
})