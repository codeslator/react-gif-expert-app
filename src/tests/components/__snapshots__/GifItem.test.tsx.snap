// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifItem /> component. Should to match with the snapshot. 1`] = `
<div>
  <div
    class="h-full p-card p-component"
    data-pc-name="card"
    data-pc-section="root"
  >
    <div
      class="p-card-header"
      data-pc-section="header"
    >
      <img
        alt="Pokemon"
        src="https://pokemon.com/bulbasaur.jpg"
      />
    </div>
    <div
      class="p-card-body"
      data-pc-section="body"
    >
      <div
        class="p-card-title"
        data-pc-section="title"
      >
        Pokemon
      </div>
    </div>
  </div>
</div>
`;
